front-end/src/Pages/ReferenceSets/New.elm

```diff
@@ -536,48 +1028,380 @@ newPdbCodeListView params =
+                            (State.biolUnit
+                                (BiolUnit.pdb
+                                    Pdb.pdbCode
+                                )
+                            )
+                        )
+                    )
+                )
+            |> with DFIRE2Results.total
+        )
+
+
+type alias RosettaTotal =
+    { pdbCode : String
+    , rosettaTotal : Maybe Float
+    }
+
+
+type alias RosettaTotalRemoteData =
+    RemoteData (Graphql.Http.Error (List RosettaTotal)) (List RosettaTotal)
+
+
+rosettaTotalQuery : Set String -> SelectionSet (List RosettaTotal) RootQuery
+rosettaTotalQuery pdbCodeList =
+    Query.preferredRosettaSubset
+        (\optionals -> { optionals | stateNumber = Absent })
+        { codes =
+            Set.toList pdbCodeList
+                |> List.map String.toLower
+        }
+        (SelectionSet.succeed RosettaTotal
+            |> with
+                (SelectionSet.map (Maybe.withDefault "Unknown PDB")
+                    (unwrapMMSS
+                        (RosettaResults.state
+                            (State.biolUnit
+                                (BiolUnit.pdb
+                                    Pdb.pdbCode
+                                )
+                            )
+                        )
+                    )
+                )
+            |> with RosettaResults.totalScore
+        )
+
+
+type alias Aggrescan3DTotal =
+    { pdbCode : String
+    , aggrescan3DTotal : Maybe Float
+    }
+
+
+type alias Aggrescan3DTotalRemoteData =
+    RemoteData (Graphql.Http.Error (List Aggrescan3DTotal)) (List Aggrescan3DTotal)
+
+
+aggrescan3DTotalQuery : Set String -> SelectionSet (List Aggrescan3DTotal) RootQuery
+aggrescan3DTotalQuery pdbCodeList =
+    Query.preferredAggrescan3dSubset
+        (\optionals -> { optionals | stateNumber = Absent })
+        { codes =
+            Set.toList pdbCodeList
+                |> List.map String.toLower
+        }
+        (SelectionSet.succeed Aggrescan3DTotal
+            |> with
+                (SelectionSet.map (Maybe.withDefault "Unknown PDB")
+                    (unwrapMMSS
+                        (Aggrescan3DResults.state
+                            (State.biolUnit
+                                (BiolUnit.pdb
+                                    Pdb.pdbCode
+                                )
+                            )
+                        )
+                    )
+                )
+            |> with Aggrescan3DResults.totalValue
+        )
+
+
+
+-- }}}
+-- {{{ Default Reference Sets
+
+
+top500 : NewReferenceSet
+top500 =
+    { id = "top-500-subset"
+    , name = "Top 500 Subset"
+    , description =
+        """A set of high-quality structures defined by the Richardson lab. This is a
+        subset of the structures in top500, as the metrics fail to run for all the
+        structures files in the set. Uses the preferred biological unit as defined by
```

Comment:
"""A set of high-quality structures defined by the Richardson lab. This is a
        subset of the structures in top500, as the metrics fail to run for some of the
        structure files in the set. Uses the preferred biological unit as defined by
        PDBe.
        """